testplan/testing/multitest/result.py

```diff
@@ -99,7 +101,23 @@ def __exit__(self, exc_type, exc_value, tb):
 assertion_state = threading.local()
 
 
-def assertion(func):
+def mark(func: Callable) -> Callable:
+    """
+    Sets the decorated function's filepath and line-range in assertion state.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        filepath = inspect.getfile(func)
+        lines, start = inspect.getsourcelines(func)
+        assertion_state.filepath = filepath
+        assertion_state.line_range = range(start, start + len(lines))
```

Comment:
file_path and line_range need to be managed as if they would be in a stack, so if there were something in thise attributes they need to be reset to that after the function call. As without that:

if you have an helper call chain like this

```
testcase -> marked_helper1 -> non_marked_helper -> marked_helper2 -> assertion1 
                                                \
                                                 -> assertion2

```
assertion 1 will correctly reported in marked_helper2 while assertion 1 will be reported in non_marked_helper as the frame walk will not see the marked helper in file_path, line_range